src/wp-admin/includes/dashboard.php

```diff
@@ -1555,3 +1558,251 @@ function wp_welcome_panel() {
+						 *
+						 * @since CP 1.0.0
+						 *
+						 * @param string $votesCount Up Voted of petition
+						 * @param string $link URL to the petitions website page
+						 * @param string $title Title of Petition
+						 * @param string $author Name of Petition Lead
+						 * @param string $status Status of petition 
+						 * @param string $createdTime Timestamp of the petition request
+						 * @param string $text_domain Text domain
+						 */
+						$votesCount = $value['votesCount'];
+						$link = $value['link'];
+						$title = $value['title'];
+						$author = $value['createdBy'];
+						$status = $value['status'];
+						$createdTime = $value['createdAt'];
+
+						cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime );
+
+					}
+				}
+
+				/** 
+				 * Loop over array to return list content for the Recent petitions.
+				 * 
+				 * @since 1.0.0
+				 * @return HTML
+				 */
+				if ( $list_item == 'recent' ) {
+					foreach( $recent['data'] as $key => $value ) {
+
+						/**
+						 * Display the ClassicPress Most Wanted(voted) petitions.
+						 *
+						 * @since CP 1.0.0
+						 *
+						 * @param string $votesCount Up Voted of petition
+						 * @param string $link URL to the petitions website page
+						 * @param string $title Title of Petition
+						 * @param string $author Name of Petition Lead
+						 * @param string $status Status of petition 
+						 * @param string $createdTime Timestamp of the petition request
+						 */
+						$votesCount = $value['votesCount'];
+						$link = $value['link'];
+						$title = $value['title'];
+						$author = $value['createdBy'];
+						$status = $value['status'];
+						$createdTime = $value['createdAt'];
+
+						cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime );
+
+					}
+				}
+
+				/** 
+				 * Loop over array to return list content for the Most Wanted(voted) petitions.
+				 * 
+				 * @since 1.0.0
+				 * @return HTML
+				 */
+				if ( $list_item == 'most-wanted' ) {
+					foreach( $most_wanted['data'] as $key => $value ) {
+
+						/**
+						 * Display the ClassicPress Most Wanted(voted) petitions.
+						 *
+						 * @since CP 1.0.0
+						 *
+						 * @param string $votesCount Up Voted of petition
+						 * @param string $link URL to the petitions website page
+						 * @param string $title Title of Petition
+						 * @param string $author Name of Petition Lead
+						 * @param string $status Status of petition 
+						 * @param string $createdTime Timestamp of the petition request
+						 */
+						$votesCount = $value['votesCount'];
+						$link = $value['link'];
+						$title = $value['title'];
+						$author = $value['createdBy'];
+						$status = $value['status'];
+						$createdTime = $value['createdAt'];
+
+						cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime );
+
+					}
+				}
```

Comment:
... and instead of repeating this section of code 3 times, we can use `$json[ $list_item ]`.

Reply:
I would also rename this data array from `$json` to `$petitions`.  This should fit well with the changes to move `json_decode` to before the `set_transient` call.